File: 2022/1/python/main.py
with open("../input.txt", "r") as f:
    lines = f.readlines()

class Elf():
    def __init__(self, number):
        self.number = number
        self.calories = 0

elves = []
elf = Elf(0)

for line in lines:
    line = line.strip()
    if not line: 
        elves.append(elf)
        elf = Elf(len(elves)-1)
    else:
         elf.calories += int(line)

elves.sort(key=lambda x: -x.calories)
top_three_elves = elves[:3]
top_three_sum = sum(map(lambda x: x.calories, top_three_elves))

print("Top three elves:")
for e in top_three_elves:
    print(f"    Elf #{e.number} has {e.calories} calories")
print(f"Together they are carrying {top_three_sum} calories")

File: 2022/1/rust/src/main.rs
use std::fs;

struct Elf {
    number: u64,
    calories: u64,
}

fn main() {
    println!("Hello, world!");
    let contents = fs::read_to_string("../input.txt").expect("Couldn't read input file");
    let mut elf = Elf{ number: 0, calories: 0};
    let mut elves = Vec::new();
    for mut line in contents.lines() {
        line = line.trim();
        if !line.is_empty() {
            elf.calories += line.parse::<u64>().unwrap();
        } else {
            elves.push(elf);
            elf = Elf{ number: elves.len() as u64, calories: 0}
        }
    }

    elves.sort_by(|a, b| b.calories.cmp(&a.calories));

    println!("Top three elves:");
    let mut sum : u64 = 0;
    for n in 0..3 {
        sum += elves[n].calories;
        println!("    Elf #{} has {} calories", elves[n].number, elves[n].calories);
    }
    println!("Together they are carrying {} calories", sum);
}
